pyensembl/genome.py

```diff
@@ -0,0 +1,848 @@
+from .exon import Exon
+from .gene import Gene
+from .gtf import GTF
+from .sequence_data import SequenceData
+from .transcript import Transcript
+
+class Genome(object):
+    """
+    Bundles together the genomic annotation and sequence data associated with
+    a particular genomic database source (e.g. a single Ensembl release) and
+    provides a wide variety of helper methods for accessing this data.
+    """
+    def __init__(self,
+                 reference_name,
+                 genome_source,
+                 name=None,
+                 version=None,
+                 only_human=False,
+                 auto_download=False,
+                 local_fasta_filename_func=None,
+                 require_ensembl_ids=True):
+        self.reference_name = reference_name
+        self.genome_source = genome_source
+        self.name = name
+        self.version = version
+        self.only_human = only_human
+        self.auto_download = auto_download
+
+        # GTF object wraps the source GTF file from which we get
+        # genome annotations. Presents access to each feature
+        # annotations as a pandas.DataFrame.
+        self.gtf = GTF(
+            genome_source,
+            auto_download=auto_download)
+
+        # Database object turns the GTF dataframes into sqlite3 tables
+        # and wraps them with methods like `query_one`
+        self.db = Database(gtf=self.gtf, auto_download=auto_download)
+
+        # get the path for the cDNA FASTA file containing
+        # this genome database's transcript sequences
+        transcript_sequences = None
+        if genome_source.transcript_fasta_path:
+            transcript_sequences = SequenceData(
+                genome_source=genome_source,
+                fasta_type="transcript",
+                local_filename_func=local_fasta_filename_func,
+                require_ensembl_ids=require_ensembl_ids,
+                auto_download=auto_download)
+        self.transcript_sequences = transcript_sequences
+
+        protein_sequences = None
+        if genome_source.protein_fasta_path:
+            protein_sequences = SequenceData(
+                genome_source=genome_source,
+                fasta_type="protein",
+                local_filename_func=local_fasta_filename_func,
+                require_ensembl_ids=require_ensembl_ids,
+                auto_download=auto_download)
+        self.protein_sequences = protein_sequences
+
+        self.logger = logging.getLogger()
+        self.logger.setLevel(logging.INFO)
+
+    def __str__(self):
+        return ("Genome(name=%s, version=%s, reference_name=%s, "
+                "only_human=%s, genome_source=%s)") % (
+            self.name,
+            self.version,
+            self.reference_name,
+            self.only_human,
+            self.genome_source)
+
+    def __repr__(self):
+        return str(self)
+
+    def __eq__(self, other):
+        return (
+            other.__class__ is Genome and
```

Comment:
Totally valid thought experiment. I've been wondering about that: also in the case of using `Genome` for mice rather than expanding `EnsemblRelease` to accommodate mice.

Maybe `EnsemblRelease.as_genome()` would be helpful here? And then that resultant object would be equal to the other?